### PR TITLE
CMake: turn on PIC mode when project is building as shared library

### PR DESCRIPTION
**Pull request by axic**
Fixes 6df6d0a

### CODE REVIEW

**Comment by axic** [comment]
This seems to be working (together with https://github.com/ewasm/hera/pull/269) so I'll merge it, but please have a look.

File: libs/evm2wasm/CMakeLists.txt [hunk 1]
```diff
@@ -7,5 +7,9 @@ add_library(libevm2wasm STATIC
 )
 set_target_properties(libevm2wasm PROPERTIES OUTPUT_NAME evm2wasm)
 
+if(BUILD_SHARED_LIBS)
+  set_target_properties(libevm2wasm PROPERTIES POSITION_INDEPENDENT_CODE TRUE)
```
Comment (chfast) [inline libs/evm2wasm/CMakeLists.txt]: You probable could always enable it.

Comment (axic) [reply libs/evm2wasm/CMakeLists.txt]: Could or should?

Comment (chfast) [reply libs/evm2wasm/CMakeLists.txt]: It should not affect the performance so the configuration will be simpler (and less number of combinations). But this is not required.

Comment (axic) [reply libs/evm2wasm/CMakeLists.txt]: It does affect the performance, but not sure if that is any significant nowadays compared to the overall complexity of the code anyway.

Comment (axic) [reply libs/evm2wasm/CMakeLists.txt]: It will be a question when we try compiling this to wasm, but that is a very long way ahead :)

Comment (chfast) [reply libs/evm2wasm/CMakeLists.txt]: Many modern linux distributions build with PIC by default.

**Pull request merged**